cf/formula_terms.py

```diff
@@ -0,0 +1,2057 @@
+
+
+def _conform_units(term, var, ref_term2, ref_units):
+    '''Make sure that the units of a variable of the formula are the same
+    as the units of another formula term.
+
+    .. versionadded:: 3.8.0
+
+    :Parameters:
+
+        term: `str`
+            A term of the formula for which the units are to
+            conformed.
+
+            *Parameter example:*
+              ``term='z2'``
+
+        var: `DomainAncillary` or `Coordinate`
+            The variable for the *term* term whose units are to
+            conformed to *ref_units*.
+
+        ref_term: `str`
+            A term of the formula which defines the reference units.
+
+            *Parameter example:*
+              ``term='href'``
+
+        ref_units: `Units`
+            The units of the *ref_term* term.
+
+            *Parameter example:*
+              ``ref_units=cf.Units('1')`
+
+    :Returns:
+
+        `DomainAncillary` or `Coordinate`
+            The input *var* construct with conformed units.
+
+    '''
+    units = var.Units
+
+    if units != ref_units:
+        if not units.equivalent(ref_units):
+            raise ValueError(
+                "Terms {!r} and {!r} have incompatible units: "
+                "{!r}, {!r} ".format(
+                    ref_term, term, ref_units, units
+                )
+            )
+
+        var = var.copy()
+        var.Units = ref_units
+
+    return var
+
+
+def _check_standard_name_consistency(strict,
+                                     zlev=(None, None),
+                                     eta=(None, None),
+                                     depth=(None, None)):
+    '''Check that there are consistent sets of values for the
+    standard_names of formula terms and the computed_standard_name
+    needed in defining the ocean sigma coordinate, the ocean
+    s-coordinate, the ocean_sigma over z coordinate, and the ocean
+    double sigma coordinate, as described Appendix D: Parametric
+    Vertical Coordinates of the CF conventions.
+
+    .. versionadded:: 3.8.0
+
+    :Parameters:
+
+        zlev: `tuple`, optional
+            The 'zlev' term domain ancillary construct and its
+            construct key. If the domain ancillary construct was
+            created from default values then the construct key will be
+            `None`.
+
+        eta: `tuple`, optional
+            The 'eta' term domain ancillary construct and its
+            construct key. If the domain ancillary construct was
+            created from default values then the construct key will be
+            `None`.
+
+        depth: `tuple`, optional
+            The 'depth' term domain ancillary construct and its
+            construct key. If the domain ancillary construct was
+            created from default values then the construct key will be
+            `None`.
+
+        strict: `bool`
+            If False then allow the computation to occur all of the
+            given the 'zlev', 'eta', and 'depth' domain ancillary
```

Comment:
You're right. I think it might be less confusing to say

"If False then allow the computation to occur even if **none** of the
            given the 'zlev', 'eta', and 'depth' domain ancillary
            constructs have **a** standard name. By default if any of
            these constructs is missing a standard name then an
            exception will be raised."

I'll fix that.